### PR TITLE
RDCC-1582, Demo: Adding component name to all application logs

### PR DESCRIPTION
**Pull request by AdilOoze**
**JIRA link (if applicable)**

https://tools.hmcts.net/jira/browse/RDCC-1582

**Change description**

Bringing changes from: https://github.com/hmcts/rd-professional-api/pull/639 & fixing changes in https://github.com/hmcts/rd-professional-api/pull/641

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/WelcomeController.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 public class WelcomeController {
 
     @Value("${logging-component-name}")
-    protected static String loggingComponentName;
+    protected String loggingComponentName;
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/WelcomeController.java:31]: use private variable everywhere if this variable is not used by child class

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/WelcomeController.java]: This has been addressed in the following commit: https://github.com/hmcts/rd-professional-api/pull/649/commits/75ebfaeeb6a31383283e6a172ce5a5834c972b07

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/validator/UserCreationRequestValidator.java [hunk 2]
```diff
@@ -4,26 +4,31 @@
...
 
     private UserCreationRequestValidator() {
     }
 
-
     public static List<String> validateRoles(List<String> roles) {
 
         if (CollectionUtils.isEmpty(roles)) {
-            log.error("{}:: No user role(s) provided",loggingComponentName);
+            log.error("{}:: No user role(s) provided", loggingComponentName);
```
Comment (KOTLAPRA23) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/validator/UserCreationRequestValidator.java:23]: better remove logger.anyway throwing error right. so no need add extra code for logger

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/validator/UserCreationRequestValidator.java]: Addressed in this commit: https://github.com/hmcts/rd-professional-api/pull/649/commits/8d6014387fe7f49d589bc62bb8104ce838d8afbb

File: src/main/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtil.java [hunk 3]
```diff
@@ -47,16 +48,15 @@
...
 
+@Component
 @Slf4j
 public class RefDataUtil {
 
     private RefDataUtil() {}
 
-    @Value("${defaultPageSize}")
-    public static final int DEFAULTPAGESIZE = 10;
+    public static int defaultPageSize;
```
Comment (KOTLAPRA23) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtil.java:57]: make private instead of public

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtil.java]: Addressed in this commit: https://github.com/hmcts/rd-professional-api/pull/649/commits/7c8443e1fc31e49bd9979dbf419c336f5e5d034a

**Pull request merged**